example/spec/helpers/kinesisHelpers.js

```diff
@@ -157,10 +159,17 @@ async function getShardIterator(streamName) {
  *
  * @param  {string} shardIterator - Kinesis stream shard iterator.
  *                                  Shard iterators must be generated using getShardIterator.
- * @returns {Promise}              - kinesis GetRecords promise
+ * @returns {Array}               - Array of records from kinesis stream.
  */
-async function getRecords(shardIterator) {
-  return kinesis.getRecords({ ShardIterator: shardIterator }).promise();
+async function getRecords(shardIterator, records = []) {
+  const data = await kinesis.getRecords({ ShardIterator: shardIterator }).promise();
+  records.push(...data.Records);
```

Comment:
The kinesis limit default is 10,000 records. https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Kinesis.html#getRecords-property

But since we're tearing down and building the streams during testing, the actual limit is closer to 2.   To protect against overflow on js params, I could add a limit key to this call, or add the records to the array differently.